src/x/net/http/errors.go

```diff
@@ -92,25 +92,30 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 		fn(&o)
 	}
 
+	statusCode := getStatusCode(err)
+	if o.response == nil {
+		w.Header().Set(HeaderContentType, ContentTypeJSON)
+		w.WriteHeader(statusCode)
```

Comment:
Why in one case it is `Header().Set(....)`, and in another - `WriteHeader(...)`, what is the difference here?

Reply:
`WriteHeader` is deceiving, it means write all the headers and a status code.  Header().Set is preparing a single header to be written when `WriteHeader` is called.